pkg/agent/controller/trafficcontrol/controller.go

```diff
@@ -0,0 +1,554 @@
+			"erspan_ver":                          strconv.Itoa(int(version)),
+			"key":                                 strconv.Itoa(1),
+		}
+		if version == 1 {
+			if index != nil {
+				externalIDs["erspan_idx"] = strconv.Itoa(int(*index))
+			}
+		} else if version == 2 {
+			externalIDs["erspan_dir"] = strconv.Itoa(int(*dir))
+			externalIDs["erspan_hwid"] = strconv.Itoa(int(*hardwareID))
+		}
+		c.ovsPortUpdateMutex.Lock()
+		defer c.ovsPortUpdateMutex.Unlock()
+		portUUID, err := c.ovsBridgeClient.CreateTunnelPortExt(portName,
+			ovsconfig.ERSpanTunnel, 0, false, "", remoteIP, "", externalIDs)
+		if err != nil {
+			return 0, err
+		}
+		ofPortTmp, err = c.ovsBridgeClient.GetOFPort(portName, false)
+		if err != nil {
+			return 0, err
+		}
+		itf := interfacestore.NewTunnelInterface(portName, ovsconfig.ERSpanTunnel, nil, false)
+		itf.OVSPortConfig = &interfacestore.OVSPortConfig{PortUUID: portUUID, OFPort: ofPortTmp}
+		c.interfaceStore.AddInterface(itf)
+		ofPort = uint32(ofPortTmp)
+
+		klog.V(2).InfoS("Create ERSPAN port", "remoteIP", remoteIP, "version", version, "index", index,
+			"dir", dir, "hardwareID", hardwareID)
+	}
+	return
+}
+
+func (c *Controller) syncTrafficControl(tcName string) (err error) {
+	startTime := time.Now()
+	defer func() {
+		klog.V(4).InfoS("Finished syncing TrafficControl", "TrafficControl", tcName, "durationTime", time.Since(startTime))
+	}()
+
+	tc, err := c.trafficControlLister.Get(tcName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			tcState, exist := c.getTcState(tcName)
+			if !exist {
+				return nil
+			}
+			if err := c.uninstallTrafficControl(tcName, tcState); err != nil {
+				return err
+			}
+			return nil
+		}
+		return err
+	}
+
+	tcState, exist := c.getTcState(tcName)
+
+	if !exist {
+		tcState = c.createTcState(tcName)
+		var returnPort, targetPort uint32
+		// ReturnPort should only be set for Redirect action.The validation webhook checked it.
+		if tc.Spec.ReturnPort != nil {
+			if returnPort, err = c.getDevicePort(tc.Spec.ReturnPort); err != nil {
+				return err
+			}
+		}
+		if targetPort, err = c.getDevicePort(&tc.Spec.TargetPort); err != nil {
+			return err
+		}
+		if err := c.ofClient.InstallTrafficControlCommonFlows(targetPort, returnPort); err != nil {
+			return err
+		}
+		tcState.targetPort = targetPort
+		tcState.dstOfOutPort = returnPort
+	}
+	var pods []*v1.Pod
+	if pods, err = c.filterPods(&tc.Spec.AppliedTo); err != nil {
+		return err
+	}
+	klog.V(4).InfoS("Filtering Pods", "localPodsNum", len(pods), "appliedTo", tc.Spec.AppliedTo)
+	if len(pods) == 0 {
+		return nil
+	}
+
+	var podOfPorts []uint32
+	for _, pod := range pods {
+		podItem := strings.Join([]string{pod.Namespace, pod.Name}, "/")
+		tcState.pods.Insert(podItem)
+
+		podInterfaces := c.interfaceStore.GetContainerInterfacesByPod(pod.Name, pod.Namespace)
+		if len(podInterfaces) == 0 {
+			klog.Warningf("Container interfaces not found, Pod: %s-%s", pod.Namespace, pod.Name)
```

Comment:
This could just be `InfoS`. In some cases Pod interface may have not been created yet but it's normal.
And structured logging just uses Info for warining.

Reply:
Done.